drivers/npmx/npmx_shell.c

```diff
@@ -1379,6 +1379,24 @@ static int buck_gpio_get(const struct shell *shell, size_t argc, char **argv,
 	return 0;
 }
 
+static bool dt_pins_collision_check(const struct shell *shell, int8_t gpio_idx)
```

Comment:
I would put that function at the beginning of the file, right after `check_error_code`, because it is used in three different parts of npmx commands (buck, ldsw and leds).